### ISSUE TITLE
fix ordering of dmvt args (put df third) so providing unnamed args works

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#1014]
Currently reparameterization of dmvt (and therefore any use of dmvt that doesn't name args) doesn't work because `df` is after `scale`, so matching of args tries to set `sigma` to be the `df` value. Haven't looked closely, but I think simply putting `df` as the third arg (after `prec`) will fix this.

**Comment by paciorek** [comment]
Also should we be able to leave `scale_param` out of the Cholesky version of `dwish`?
```
> code = nimbleCode({
+     W[1:3,1:3] ~ dwish(cholesky = U[1:3,1:3], df = 5)
+ })
> m = nimbleModel(code, inits = list(U = diag(3)))
defining model...
Error: bad parameters for distribution dwish(df = 5, cholesky = U[1:3, 1:3]). (No available re-parameterization found.)
```

**Comment by paciorek** [comment]
In answer to my own question, we don't have default parameter values in BUGS code, so makes sense that if you use `cholesky` in either `dmnorm` or `dwish`, you need to explicitly give `prec_param`, `scale_param`, etc.

**Comment by paciorek** [comment]
Fixed by PR #1027.
